src/common/btree.c

```diff
@@ -1406,6 +1581,80 @@ btr_probe_valid(dbtree_probe_opc_t opc)
 		opc == BTR_PROBE_GE || opc == BTR_PROBE_LE);
 }
 
+static enum btr_probe_rc
+btr_probe_embedded(struct btr_context *tcx, dbtree_probe_opc_t probe_opc, uint32_t intent,
+		   d_iov_t *key, char hkey[DAOS_HKEY_MAX])
+{
+	struct btr_record *rec = &tcx->tc_record;
+	int                rc;
+	int                rc2;
+	int                cmp       = BTR_CMP_EQ; /** for FIRST/LAST probe, this doesn't matter */
+	int                hash_flag = 0;
+
+	D_ASSERTF(!btr_is_int_key(tcx), "Embedded root is incompatible with integer keys\n");
+
+	/** Use the fake record since only the user allocated part is stored
+	 *  in the tree root.
+	 */
+	rec->rec_off = tcx->tc_tins.ti_root->tr_node;
+
+	rc = PROBE_RC_OK;
+	if (probe_opc & BTR_PROBE_SPEC) {
+		if (key != NULL) {
+			/** Simple case */
+			cmp = btr_key_cmp(tcx, rec, key);
+		} else {
+			/** Restoring from anchor.  For direct key, the decoded
+			 * key will have been passed in. Otherwise, only the
+			 * hash key is passed in so we need to compute the hash
+			 * key for the embedded record
+			 */
+			rc2 = btr_embedded_create_hash(tcx, true);
+			if (rc2 != 0)
```

Comment:
[minor] this should convert to enum btr_probe_rc type.

Reply:
unlikely to happen but yes, we should.   Good catch!